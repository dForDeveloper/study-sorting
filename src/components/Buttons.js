import React from 'react';

function Buttons(props) {
  return (
    <footer className="footer">
      {!props.demoStarted && 
        <div className="btn-container">
          <button className="btn-start" onClick={props.startDemo}>
            start
          </button>
        </div>
      }
      {props.demoStarted && 
        <div className="btn-container">
          {props.currentStep > 1 &&
            <button className="btn-back" onClick={() => props.goToStep(-1)}>
              back
            </button>
          }
          <button className="btn-restart" onClick={props.restartDemo}>
            restart
          </button>
          {props.currentStep !== props.lastStep &&
            <button className="btn-next" onClick={() => props.goToStep(1)}>
              next
            </button>
          }
        </div>
      }
    </footer>
  )
}

export default Buttons;