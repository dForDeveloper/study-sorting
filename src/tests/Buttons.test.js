import React from 'react';
import Buttons from '../components/Buttons';
import { shallow } from 'enzyme';

const startAlgorithmMock = jest.fn();
const goForwardMock = jest.fn();

describe('Buttons', () => {
  it('should match the snapshot when is showStartButton is false', () => {
    let wrapper = shallow(
      <Buttons
        showStartButton={false}
        startAlgorithm={startAlgorithmMock}
        goForward={goForwardMock}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});