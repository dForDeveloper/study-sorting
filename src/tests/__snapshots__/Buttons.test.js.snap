// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should match the snapshot when is showStartButton is false 1`] = `
<footer
  className="footer"
>
  <div>
    <button
      className="footer--btn"
      onClick={[MockFunction]}
    >
      start
    </button>
  </div>
</footer>
`;
